fix: Issue #631 raise an error when trying to extract a file (.zip_)

WarrenU opened the pull request:
Addresses issue #631 Unzipping a malformed/invalid zip file by raising an error if the extension is wrong.
- Added `isExensionExtractable` function to check against file types we can extract from the `xtractr` package. (return bool if we can extract or not)
- Removed internal unzip function, as `xtractr` can handle unzipping zip file.
- example log: `time=2025-02-21T10:53:00.459-08:00 level=ERROR msg="Error unexpected file extension type: .zip_" error="unsupported operation"`

Trial screenshot below (No progress in bottom left as also mentioned in issue):
![image](https://github.com/user-attachments/assets/e601ea6a-5e7f-42b1-afd5-8196c5e4afcc)

❓ This is my first PR and I'm a brand new beginner to superfile.
Where would I see these logged errors? I ran it with VScode and put some breakpoints in to ensure the code I was hitting was working as expected. And it made some `__debug_bin3073481771.exe` files. I'm running it on windows. But not sure how to introspect that file or where to see the errors.

jachewz commented:
sort of related but is it a good idea to remove the ".zip" case since extractCompressFile() can handle".zip" files as well? 
https://github.com/yorukot/superfile/blob/0cd46c47df18cada14966edd1756f9ca23136374/src/internal/handle_file_operations.go#L534-L548

Reduces the need to maintain the in-house unzip() function.

lazysegtree commented:
@WarrenU `slog.Error` goes to superfile log file whose location you can see by running `spf path-list` .

lazysegtree commented on src/internal/string_function.go:
```diff
@@ -154,6 +156,22 @@ func isBufferPrintable(buffer []byte) bool {
 
+// isValidFileExtension checks if a string is a valid file extension.
+// Returns nil if valid, otherwise returns an error with a descriptive message.
+func isValidFileExtension(ext string) error {
```
This function should return a bool, not an error. 
error denotes failures in execution. If you give an invalid extension to this function, it successfully detects that its invalid, and return false. There is no error.

lazysegtree commented on src/internal/handle_file_operations.go:
```diff
@@ -519,6 +519,12 @@ func (m *model) extractFile() {
 	var err error
 	panel := &m.fileModel.filePanels[m.filePanelFocusIndex]
 	ext := strings.ToLower(filepath.Ext(panel.element[panel.cursor].location))
+	err = isValidFileExtension(ext)
```
We dont wanna check if the extension is valid, but if it can be extracted or not. I dont think `.jpg` can be extracted. We should replace this with `isExensionExtractable` and just check for a few supported extenstions `.zip`, .`tar.gz`, `.rar` whichever we support.

WarrenU replied:
Updated to `isExensionExtractable ` that makes sense and I agree.

lazysegtree commented:
> sort of related but is it a good idea to remove the ".zip" case since extractCompressFile() can handle".zip" files as well?

@jachewz If it can handle, then we should remove it.  Then we dont need the `unzip` function and we can delete that .

WarrenU commented:
> > sort of related but is it a good idea to remove the ".zip" case since extractCompressFile() can handle".zip" files as well?
> 
> @jachewz If it can handle, then we should remove it. Then we dont need the `unzip` function and we can delete that .

I removed the unzip, and opted to use xtractr default. I also added an explicit slog call for any error it raises.

Good mention @jachewz

lazysegtree commented:
I would like us a to do a bit of testing that it can handle .tar.gz , .rar and most of these extensions.  I will try to test it out sometimes.

WarrenU commented:
.rar, .tar.gz, .zip are working. ✅

yorukot commented:
I just discovered during testing that if there is a directory in the decompression, permission denied will occur, so I added:
```diff
	x := &xtractr.XFile{
		FilePath:  src,
		OutputDir: dest,
+		FileMode:  0644,
+		DirMode:   0755,
	}
```

Now it seems that it should be merged.
